### PR TITLE
[WIP] Add possible french tags for card auto-fill

### PR DESCRIPTION
**Pull request by Crocmagnon**
I added some values that might be used by french developers in their forms.

Discussion started in #521

### CODE REVIEW

**Comment by Crocmagnon** [comment]
@kspearrin, I can't test if the filling works properly now because the extension (loaded via the temporary extensions in the `about:debugging` page) says this:
screeenshot

In English : "Impossible to auto-fill the selected site on this page. Please copy/paste you username and/or password" (which by the way doesn't seem an appropriate message for a card auto-fill fail 😉)

File: src/services/autofill.service.ts [hunk 1]
```diff
@@ -426,29 +426,32 @@ export default class AutofillService implements AutofillServiceInterface {
...
                 // ref https://developers.google.com/web/fundamentals/design-and-ux/input/forms/
                 if (!fillFields.cardholderName && this.isFieldMatch(f[attr],
-                    ['cc-name', 'card-name', 'cardholder-name', 'cardholder', 'name'],
+                    ['cc-name', 'card-name', 'cardholder-name', 'cardholder', 'name', 'nom'],
                     ['cc-name', 'card-name', 'cardholder-name', 'cardholder'])) {
                     fillFields.cardholderName = f;
                 } else if (!fillFields.number && this.isFieldMatch(f[attr],
-                    ['cc-number', 'cc-num', 'card-number', 'card-num', 'number', 'cc', 'cc-no', 'card-no'],
+                    ['cc-number', 'cc-num', 'card-number', 'card-num', 'number', 'cc', 'cc-no', 'card-no', 'numero-carte',
+                        'carte', 'carte-credit'],
```
Comment (kspearrin) [inline src/services/autofill.service.ts]: `carte-credit` and `numero-carte` could probably be added to the contains checks i would imagine. We could probably add the English `credit-card` value too now that i see we are missing that.

File: src/services/autofill.service.ts [hunk 2]
```diff
@@ -426,29 +426,32 @@ export default class AutofillService implements AutofillServiceInterface {
...
                     ['cc-number', 'cc-num', 'card-number', 'card-num', 'cc-no', 'card-no'])) {
                     fillFields.number = f;
                 } else if (!fillFields.exp && this.isFieldMatch(f[attr],
-                    ['cc-exp', 'card-exp', 'cc-expiration', 'card-expiration', 'cc-ex', 'card-ex', 'card-expire'],
-                    [])) {
+                    ['cc-exp', 'card-exp', 'cc-expiration', 'card-expiration', 'cc-ex', 'card-ex', 'card-expire',
```
Comment (kspearrin) [inline src/services/autofill.service.ts]: Why did you get rid of the contains exclusion? There was an empty array being passed.

Comment (Crocmagnon) [reply src/services/autofill.service.ts]: Didn't read well the `isFieldMatch` method.
I thought this was the remaining of a parameter used back then but I re-inserted it in the last commit.

File: src/services/autofill.service.ts (hunk 1 above)
Comment (Crocmagnon) [reply src/services/autofill.service.ts]: Did that.

**Comment by Crocmagnon** [comment]
No, the form that I submitted to you in #521 was the first payment form I inspected.
I just made assumptions based on the english values you already use and values I thought wouldn't give too much false positives.

**Pull request merged**